recipe/meta.yaml

```diff
@@ -22,29 +22,31 @@ requirements:
     - {{ compiler('c') }}
     - {{ compiler('cxx') }}
   host:
-    - python
+    - python >= 3.8
     - pip <22.1
     - numpy {{ numpy }}
     - cython
     - packaging
 
   run:
-    - python
-    - scipy >=1.0.0
-    - {{ pin_compatible('numpy', lower_bound='1.18') }}
+    - python >= 3.8
+    - scipy >=1.5.0
+    - {{ pin_compatible('numpy', lower_bound='1.19') }}
     - matplotlib-base >=1.5.1
     - seaborn
     - netcdf4 >=1.0
     - biopython >=1.71
-    - networkx >=1.0
+    - networkx >=2.0
     - griddataformats >=0.4.0
     - joblib >=0.12
     - gsd >=1.9.3  # [win]
     - gsd >=1.4.0  # [not win]
     - mmtf-python >=1.0.0
     - scikit-learn
+    - tidynamics >= 1.0.0
```

Comment:
tidynamics had been forgotten, should be there for a full feature installation

Reply:
So my solution / guide going forward here is that we do the following: include all optional deps which aren't converter related. With the exception of chemfiles (purely because we have a few more teething issues with it at the moment and I want to make the pins for chemfiles a bit more explicit I think)

With this option we would add h5py too here.

Reply:
I looked at setup.py and there tidynamics was listed under the [analysis] option and IIRC we are installing other stuff that’s listed there. 

Tidynamics is lightweight as it only requires numpy.

I agree that we should have some guidelines what to do with optional deps.

Reply:
Excluding converter deps sounds like a feasible way forward.

Reply:
I think netcdf4 has h5py as a dependency anyway... so including it would not cost us more. So I agree to adding h5py/hdf5.

Let's stick with the arbitrary, cough, "historical" approach for 2.2.0 and then have a discussion how we want to do optional deps at all levels (code, testing, pip, conda).

Reply:
I also admit to being confused to look where we authoritatively declare dependencies — I guess it's not setup.py. In principle, the GH actions files for the tests are functionally the most relevant.

Reply:
So our main dependencies are those listed in `install_requires` under setup.py. `extra_requires` is meant to be optional deps that can be installed via pypi I believe (so openmm and rdkit wouldn't be able to be pulled).

Reply:
Yes, I get that. But as you say, it leaves out anything that's not on pypi. Ultimately, https://github.com/MDAnalysis/mdanalysis/blob/develop/.github/actions/setup-deps/action.yaml is really the list of packages that work because these are the ones we test.

Reply:
Ah ok yeah, there was talk at some point to have an environmemt.yaml or requirements.txt somewhere that we can hold somewhere. I think we need to have a big discussion about how to do that though, our current approach is.. a bit messy (every time we add something we end up having to hunt a dozen different files for places to update requirements).

Reply:
Yes, 💯  — I just don't have a good solution at hand. I suppose it also reflects the fact that installation is still quite heterogeneous and different users want different things. 

From a user perspective, really only our installation instructions (User Guide and quick install on website) need to be up-to-date and working. But perhaps we could make our life easier by keeping requirements files for pip and conda in the repo and then use `wget` or `curl` in our installation instructions for full-feature installations?